ocl/openmrsdictionary/serializers.py

```diff
@@ -0,0 +1,301 @@
+        dictionary.public_access = attrs.get(
+            'public_access', dictionary.public_access or DEFAULT_ACCESS_TYPE)
+        dictionary.default_locale = attrs.get(
+            'default_locale', dictionary.default_locale or Common.DEFAULT_LOCALE)
+        dictionary.website = attrs.get('website', dictionary.website)
+        dictionary.supported_locales = attrs.get('supported_locales').split(
+            ',') if attrs.get('supported_locales') else dictionary.supported_locales
+        dictionary.extras = attrs.get('extras', dictionary.extras)
+        dictionary.external_id = attrs.get(
+            'external_id', dictionary.external_id)
+        return dictionary
+
+
+class DictionaryCreateSerializer(DictionaryCreateOrUpdateSerializer):
+    """
+    Serializer used to update an openmrsdictionary
+    """
+    type = serializers.CharField(source='resource_type', read_only=True)
+    uuid = serializers.CharField(source='id', read_only=True)
+    id = serializers.CharField(
+        required=True,
+        validators=[
+            RegexValidator(
+                regex=NAMESPACE_REGEX)],
+        source='mnemonic')
+    short_code = serializers.CharField(source='mnemonic', read_only=True)
+    name = serializers.CharField(required=True)
+    full_name = serializers.CharField(required=False)
+    description = serializers.CharField(required=False)
+    custom_validation_schema = serializers.CharField(required=False)
+    #collection_type = serializers.CharField(required=False)
+    public_access = serializers.ChoiceField(
+        required=False, choices=ACCESS_TYPE_CHOICES)
+    default_locale = serializers.CharField(required=False)
+    supported_locales = serializers.CharField(required=False)
+    website = serializers.CharField(required=False)
+    url = serializers.CharField(read_only=True)
+    versions_url = serializers.CharField(read_only=True)
+    concepts_url = serializers.CharField(read_only=True)
+    mappings_url = serializers.CharField(read_only=True)
+    owner = serializers.CharField(source='parent_resource', read_only=True)
+    owner_type = serializers.CharField(
+        source='parent_resource_type', read_only=True)
+    owner_url = serializers.CharField(source='parent_url', read_only=True)
+    versions = serializers.IntegerField(source='num_versions', read_only=True)
+    created_on = serializers.DateTimeField(source='created_at', read_only=True)
+    updated_on = serializers.DateTimeField(source='updated_at', read_only=True)
+    created_by = serializers.CharField(read_only=True)
+    updated_by = serializers.CharField(read_only=True)
+    extras = serializers.WritableField(required=False)
+    external_id = serializers.CharField(required=False)
+    preferred_source = serializers.CharField(
+        required=True, validators=[
+            RegexValidator(
+                regex=NAMESPACE_REGEX)])
+    # active_concepts = serializers.SerializerMethodField(method_name='get_active_concepts')
```

Comment:
@mwaz Could you remove the commented out code because it is not doing anything at the moment

Reply:
resolved

Reply:
resolved